### ISSUE TITLE
Ruffle Nightly 2023-11-30 and 2023-11-29 keeps crashing; Map callback was leaked

### ISSUE THREAD

**Issue by FlashBacks1998** [ruffle-rs/ruffle#14247]
### Describe what you were doing

I was loading a scene from one of **Away3D** demos, the terrain scene. Seemingly (when the scene loads only) I keep crashing and the Ruffle program hangs and panics. At first it was an issue with Nightly 2023-11-30 kept crashing and Nightly 2023-11-29 seemed fine, but for whatever reason Nightly 2023-11-29 started to panic when the scene starts to load as well.

The _extremely_ weird thing about this bug is, it only affects both Nightly versions when the screen is _minimized_. What I mean is when I have two instances of Ruffle running (both Nightly 2023-11-29 and/or Nightly 2023-11-30 it doesn't matter) running while the scene is loading and the Window hangs and I minimize it, its a guaranteed panic.

**EDIT:** I was testing the Newgrounds upload again after I submitted it and it looks like I got another panic, Attached are the logs as well

The Newgrounds portal version can be found [here](https://www.newgrounds.com/projects/games/5244634/preview), while the [source code](http://www.mediafire.com/file/nmazldo9nplr8nr/Advanced_TerrainDemo.zip) and [compiled .swf binaries](https://www.mediafire.com/file/7xb3dhgmv31j61d/Advanced_TerrainDemo.swf/file) can be found on MediaFire.

### What does the crash message say?
Ruffle Nightly
```
Ruffle has encountered a fatal error, this is a bug.

panicked at
C:\Users\runneradmin\.cargo\git\checkouts\wgpu-53e70f8674b08dd4\4bf9c5\wgpu-core\src\resource.rs:207:13: 
Map callback was leaked

Please report this to us so that we can fix it. Thank you
```!

### Ruffle Version

Nightly 2023-11-29 to Nightly 2023-11-30

### Affected platform

Desktop app

### Operating system

Windows 10

### Additional information

The web version works completely fine, I've already tested.

**Comment by Lord-McSweeney** [comment]
This is probably gfx-rs/wgpu#4214, which we haven't yet pulled in (it should be present in v0.18).

**Comment by FlashBacks1998** [comment]
I got another error when running the example. This time it was actually in the web port of Ruffle. Attached below are the logs

# Error Info
Error name: Error
Error message: panicked at /home/runner/.cargo/git/checkouts/wgpu-53e70f8674b08dd4/fb4f9c5/wgpu-core/src/resource.rs:207:13:
Map callback was leaked
Error stack:
```
Error: panicked at /home/runner/.cargo/git/checkouts/wgpu-53e70f8674b08dd4/fb4f9c5/wgpu-core/src/resource.rs:207:13:
Map callback was leaked
    at H.n.wbg.__wbg_new_ab87fd305ed9004b (https://uploads.ungrounded.net/ruffle_wrapper/lib/core.ruffle.cc7cf63659ee97d0aa27.js:1:56255)
    at core::ops::function::FnOnce::call_once{{vtable.shim}}::h6f6a774408238ff0 (wasm://wasm/0336b8a2:wasm-function[9800]:0x9afd7a)
    at std::panicking::rust_panic_with_hook::hc20eadded6bfe687 (wasm://wasm/0336b8a2:wasm-function[6287]:0x8a06e7)
    at core::panicking::panic_fmt::h7a368385936888dc (wasm://wasm/0336b8a2:wasm-function[9605]:0x9aac5f)
    at <wgpu_core::resource::BufferMapCallback as core::ops::drop::Drop>::drop::ha2d666eddaf8eebb (wasm://wasm/0336b8a2:wasm-function[8946]:0x967aa6)
    at <T as wgpu::context::DynContext>::surface_configure::hb636f158c37f9809 (wasm://wasm/0336b8a2:wasm-function[8110]:0x930a0f)
    at wgpu::Surface::configure::hce634451e402b254 (wasm://wasm/0336b8a2:wasm-function[4146]:0x78c0a2)
    at <ruffle_render_wgpu::target::SwapChainTarget as ruffle_render_wgpu::target::RenderTarget>::resize::h4bedd4a1eacc4b88 (wasm://wasm/0336b8a2:wasm-function[10167]:0x9c4cb2)
    at <ruffle_render_wgpu::backend::WgpuRenderBackend<T> as ruffle_render::backend::RenderBackend>::set_viewport_dimensions::h2d09e2bbc6d9a1aa (wasm://wasm/0336b8a2:wasm-function[5114]:0x83bf69)
    at ruffle_core::player::Player::set_viewport_dimensions::h33f3d1b57f47ec1d (wasm://wasm/0336b8a2:wasm-function[1650]:0x4e694d)
```

# Player Info
Allows script access: false
Renderer: wgpu
Adapter Backend: Gl
Adapter Name: "ANGLE (NVIDIA, NVIDIA GeForce RTX 2080 Super with Max-Q Design (0x00001ED3) Direct3D11 vs_5_0 ps_5_0, D3D11)"
Adapter Device Type: Other
Adapter Driver Name: ""
Adapter Driver Info: ""
Enabled features: Features(TEXTURE_COMPRESSION_BC | TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES | PUSH_CONSTANTS | SHADER_UNUSED_VERTEX_OUTPUT)
Available features: Features(CLEAR_TEXTURE | MULTIVIEW)
Current limits: Limits { max_texture_dimension_1d: 16384, max_texture_dimension_2d: 16384, max_texture_dimension_3d: 2048, max_texture_array_layers: 256, max_bind_groups: 4, max_bindings_per_bind_group: 1000, max_dynamic_uniform_buffers_per_pipeline_layout: 8, max_dynamic_storage_buffers_per_pipeline_layout: 0, max_sampled_textures_per_shader_stage: 16, max_samplers_per_shader_stage: 16, max_storage_buffers_per_shader_stage: 0, max_storage_textures_per_shader_stage: 0, max_uniform_buffers_per_shader_stage: 11, max_uniform_buffer_binding_size: 16384, max_storage_buffer_binding_size: 0, max_vertex_buffers: 8, max_buffer_size: 268435456, max_vertex_attributes: 16, max_vertex_buffer_array_stride: 255, min_uniform_buffer_offset_alignment: 256, min_storage_buffer_offset_alignment: 256, max_inter_stage_shader_components: 60, max_compute_workgroup_storage_size: 0, max_compute_invocations_per_workgroup: 0, max_compute_workgroup_size_x: 0, max_compute_workgroup_size_y: 0, max_compute_workgroup_size_z: 0, max_compute_workgroups_per_dimension: 0, max_push_constant_size: 96 }
Surface quality: high
Surface samples: 4
Surface size: Extent3d { width: 550, height: 400, depth_or_array_layers: 1 }

# Page Info
Page URL: https://uploads.ungrounded.net/ruffle_wrapper/ruffleembed.html?v=1.0.67&browsermode=default&props=%7B%22ruffle%22%3A%22%2Fruffle_wrapper%2Flib%2Fruffle.js%3F1701266274%22%2C%22public_path%22%3A%22%2Fruffle_wrapper%2Flib%2F%22%2C%22swf%22%3A%22https%3A%2F%2Fuploads.ungrounded.net%2F908000%2F908569_dd2af61701310616.swf%3F1701335786%22%2C%22vars%22%3A%7B%22NewgroundsAPI_PublisherID%22%3A1%2C%22NewgroundsAPI_SandboxID%22%3A%22656852eea6565%22%2C%22NewgroundsAPI_SessionID%22%3A%22DhYzPsSOGVWUkNrB8fbFb32a0b728b992f7dd8a63fe1bab88d5d074c39b2WFYr%22%2C%22NewgroundsAPI_UserName%22%3A%22FlashBacks1998%22%2C%22NewgroundsAPI_UserID%22%3A19218971%2C%22ng_username%22%3A%22FlashBacks1998%22%7D%2C%22width%22%3A970%2C%22height%22%3A700%2C%22icon%22%3A%22https%3A%2F%2Fpicon.ngfiles.com%2F908000%2Fflash_908569_card.webp%3Ff1701335790%22%2C%22warnOnUnsupportedContent%22%3Afalse%7D
SWF URL: https://uploads.ungrounded.net/908000/908569_dd2af61701310616.swf?1701335786

# Browser Info
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36
Platform: Win32
Has touch support: false

# Ruffle Info
Version: 0.1.0
Name: nightly 2023-11-29
Channel: nightly
Built: 2023-11-29T00:14:42.496Z
Commit: 9e146c0f49d0067e4f29fd8c647bb3275b68c60f
Is extension: false

# Metadata
width: 960
height: 700
frameRate: 44
numFrames: 1
swfVersion: 33
backgroundColor: #000000
isActionScript3: true
uncompressedLength: 4502280

**Comment by n0samu** [comment]
See also: #13408, #14051, etc

**Comment by n0samu** [comment]
Fixed by #13732
